### PR TITLE
CI: Use micromamba

### PR DESCRIPTION
**Pull request by martinRenou**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -21,20 +21,40 @@ jobs:
...
 
     - name: Mamba install dependencies
       shell: bash -l {0}
-      run: mamba install mamba python=${{ matrix.python-version }} "conda-build>=3.20" colorama pip ruamel ruamel.yaml rich pytest
+      run: |
+        export MAMBA_ROOT_PREFIX=~/micromamba
+        export MAMBA_EXE=$(pwd)/micromamba
+        . $MAMBA_ROOT_PREFIX/etc/profile.d/mamba.sh
+
+        mamba install mamba python=${{ matrix.python-version }} "conda-build>=3.20" colorama pip ruamel ruamel.yaml rich pytest -c conda-forge
```
Comment (wolfv) [inline .github/workflows/main.yml:42]:
```suggestion
        ./micromamba install mamba python=${{ matrix.python-version }} "conda-build>=3.20" colorama pip ruamel ruamel.yaml rich pytest -c conda-forge
```

**Comment by martinRenou** [comment]
@wolfv it looks like pip does not see the installed packages, any idea?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -21,20 +21,42 @@ jobs:
...
+        ./micromamba create -f ./tests/env.yml -y
 
     - name: Install boa
       shell: bash -l {0}
-      run: pip install .
+      run: |
+        export MAMBA_ROOT_PREFIX=~/micromamba
+        export MAMBA_EXE=$(pwd)/micromamba
+        . $MAMBA_ROOT_PREFIX/etc/profile.d/mamba.sh
+        ./micromamba activate test
```
Comment (wolfv) [inline .github/workflows/main.yml:50]:
```suggestion
        micromamba activate test
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -21,20 +21,42 @@ jobs:
...
+        pip install .
 
     - name: Run tests
       shell: bash -l {0}
-      run: pytest tests
+      run: |
+        export MAMBA_ROOT_PREFIX=~/micromamba
+        export MAMBA_EXE=$(pwd)/micromamba
+        . $MAMBA_ROOT_PREFIX/etc/profile.d/mamba.sh
+        ./micromamba activate test
```
Comment (wolfv) [inline .github/workflows/main.yml:60]:
```suggestion
        micromamba activate test
```

**Comment by wolfv** [comment]
yeah activation has to run through micromamba (without the dot) because it's a bash-function in the ` $MAMBA_ROOT_PREFIX/etc/profile.d/mamba.sh` script :) it's tricky! We need to make this better

**Comment by wolfv** [comment]
yayy :tada:

**Pull request merged**